crates/driver/src/domain/mempools.rs

```diff
@@ -31,35 +42,36 @@ impl Mempools {
         let auction_id = settlement.auction_id;
         let solver_name = solver.name();
 
-        let (tx_hash, _remaining_futures) = select_ok(self.0.iter().cloned().map(|mempool| {
-            async move {
-                let result = match &mempool {
-                    infra::Mempool::Boundary(mempool) => {
-                        mempool.execute(solver, settlement.clone()).await
-                    }
-                    infra::Mempool::Native(_) => {
-                        todo!("implement")
-                    }
-                };
-                observe::mempool_executed(&mempool, settlement, &result);
-                result
-            }
-            .instrument(tracing::info_span!(
-                "execute",
-                solver = ?solver_name,
-                ?auction_id,
-            ))
-            .boxed()
-        }))
-        .await?;
+        let (tx_hash, _remaining_futures) =
+            select_ok(self.mempools.iter().cloned().map(|mempool| {
+                async move {
+                    let result = match &mempool {
+                        infra::Mempool::Boundary(mempool) => {
+                            mempool.execute(solver, settlement.clone()).await
+                        }
+                        infra::Mempool::Native(mempool) => {
+                            self.submit(mempool, solver, settlement).await
+                        }
+                    };
+                    observe::mempool_executed(&mempool, settlement, &result);
+                    result
+                }
+                .instrument(tracing::info_span!(
+                    "execute",
+                    solver = ?solver_name,
+                    ?auction_id,
```

Comment:
I think we already have the auction id as part of the tracing arguments in the `/execute` handler of the `driver`.
[Example logs](https://production-6de61f.kb.eu-central-1.aws.cloud.es.io/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-24h%2Fh,to:now))&_a=(columns:!(log),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:c0e240e0-d9b3-11ed-b0e6-e361adffce0b,key:kubernetes.labels.app_kubernetes_io%2Fcomponent,negate:!t,params:(query:controller),type:phrase),query:(match_phrase:(kubernetes.labels.app_kubernetes_io%2Fcomponent:controller))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:c0e240e0-d9b3-11ed-b0e6-e361adffce0b,key:kubernetes.container_name,negate:!t,params:(query:mainnet-api-prod),type:phrase),query:(match_phrase:(kubernetes.container_name:mainnet-api-prod))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:c0e240e0-d9b3-11ed-b0e6-e361adffce0b,key:kubernetes.labels.network,negate:!f,params:(query:mainnet),type:phrase),query:(match_phrase:(kubernetes.labels.network:mainnet)))),grid:(columns:('@timestamp':(width:164))),index:c0e240e0-d9b3-11ed-b0e6-e361adffce0b,interval:auto,query:(language:kuery,query:'log:%20%22execute%22%20and%20log:%20%22settle%22'),sort:!(!('@timestamp',desc))))